packages/babel-core/src/transformation/plugin.js

```diff
@@ -12,8 +13,22 @@ export default class Plugin extends Store {
     super();
 
     this.initialized = false;
-    this.raw         = assign({}, plugin);
-    this.key         = this.take("name") || key;
+    this.raw = assign({}, plugin);
+    let specifiedName = this.take("name");
+
+    // List of invalid names.
+    if ([undefined, null, false].indexOf(specifiedName) !== -1) {
```

Comment:
Just curious, wouldn't it be easier just to check the truthy value of `specifiedName`?

something like `if (specifiedName) { doThis(); }` ? Is there any specific advantage to looking for specifically those values that are in the array?